Crash on rotation when scrolling in the media library

### Expected behavior
The app won't crash when I try to add an image from the StockLibrary.

### Actual behavior
The app occasionally crashes when I "rotate device/add image to a post" while scrolling in the media library.

### Steps to reproduce the behavior
1. Create a new post
2. Add media -> WordPress media library
3. Scroll and rotate your device at the same time (crashes from time to time)

##### Tested on [device], Android [version]
Emulator Pixel 2, Android 8.1


Note: The onViewRecycled method is called after the parent activity is destroyed.
```Process: org.wordpress.android.beta, PID: 11574
    java.lang.RuntimeException: Unable to destroy activity {org.wordpress.android.beta/org.wordpress.android.ui.media.MediaBrowserActivity}: java.lang.IllegalArgumentException: You cannot start a load for a destroyed activity
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4383)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4401)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4675)
        at android.app.ActivityThread.-wrap18(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1595)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.IllegalArgumentException: You cannot start a load for a destroyed activity
        at com.bumptech.glide.manager.RequestManagerRetriever.assertNotDestroyed(RequestManagerRetriever.java:312)
        at com.bumptech.glide.manager.RequestManagerRetriever.get(RequestManagerRetriever.java:131)
        at com.bumptech.glide.manager.RequestManagerRetriever.get(RequestManagerRetriever.java:115)
        at com.bumptech.glide.Glide.with(Glide.java:697)
        at org.wordpress.android.modules.GlideApp.with(GlideApp.java:88)
        at org.wordpress.android.util.image.ImageManager.cancelRequestAndClearImageView(ImageManager.kt:157)
        at org.wordpress.android.ui.media.MediaGridAdapter.onViewRecycled(MediaGridAdapter.java:255)
        at org.wordpress.android.ui.media.MediaGridAdapter.onViewRecycled(MediaGridAdapter.java:49)```

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java
@@ -189,6 +189,7 @@ public class MediaGridAdapter extends RecyclerView.Adapter<MediaGridAdapter.Grid
             holder.mFileTypeView.setText(fileExtension.toUpperCase(Locale.ROOT));
             int placeholderResId = WPMediaUtils.getPlaceholder(fileName);
             holder.mFileTypeImageView.setImageResource(placeholderResId);
+            mImageManager.cancelRequestAndClearImageView(holder.mImageView);
         }
         holder.mImageView.setContentDescription(mContext.getString(R.string.media_grid_item_image_desc,
                 StringUtils.notNullStr(media.getFileName())));
@@ -252,7 +253,6 @@ public class MediaGridAdapter extends RecyclerView.Adapter<MediaGridAdapter.Grid
     public void onViewRecycled(GridViewHolder holder) {
         super.onViewRecycled(holder);
         holder.mImageView.setTag(R.id.media_grid_file_path_id, null);
-        mImageManager.cancelRequestAndClearImageView(holder.mImageView);
     }
 
     public ArrayList<Integer> getSelectedItems() {

--- a/WordPress/src/main/java/org/wordpress/android/ui/people/PeopleListFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/people/PeopleListFragment.java
@@ -433,7 +433,6 @@ public class PeopleListFragment extends Fragment {
         @Override public void onViewRecycled(@NonNull ViewHolder holder) {
             super.onViewRecycled(holder);
             PeopleViewHolder peopleViewHolder = (PeopleViewHolder) holder;
-            mImageManager.cancelRequestAndClearImageView(peopleViewHolder.mImgAvatar);
         }
 
         public class PeopleViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

